### ISSUE TITLE
takeScreenshot() of a web element does not work

### ISSUE THREAD

**Issue by tepez** [angular/protractor#2892]
I tried taking a screenshot of a specific web element (as opposed to the entire page).

* When I tried doing `$(...).takeScreenshot()` I got a `$(...).takeScreenshot is not a function` error.
This was solved by adding `takeScreenshot` to the [`WEB_ELEMENT_FUNCTIONS`](https://github.com/angular/protractor/blob/master/lib/element.js#L5) array.

I can submit a PR if it was indeed unintentionally left out.

* After adding it I got the following error:
```
- Failed: GET /session/fa1f62f1-1f55-4f92-8175-4e13d15d5c66/element/0/screenshot
Build info: version: '2.48.2', revision: '41bccdd', time: '2015-10-09 19:59:12'
System info: host: 'home', ip: '192.168.80.1', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.8.0_66'
Driver info: driver.version: unknown
Build info: version: '2.48.2', revision: '41bccdd', time: '2015-10-09 19:59:12'
System info: host: 'home', ip: '192.168.80.1', os.name: 'Windows 7', os.arch: 'amd64', os.version: '6.1', java.version: '1.8.0_66'
Driver info: driver.version: unknown
at new bot.Error (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\node_modules\selenium-webdriver\lib\atoms\error.js:108:18)
at Object.bot.response.checkResponse (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\node_modules\selenium-webdriver\lib\atoms\response.js:109:9)
at C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\node_modules\selenium-webdriver\lib\webdriver\webdriver.js:379:20
at [object Object].promise.Promise.goog.defineClass.invokeCallback_ (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:1337:14)
at [object Object].promise.ControlFlow.goog.defineClass.goog.defineClass.abort_.error.executeNext_.execute_ (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2776:14)
at [object Object].promise.ControlFlow.goog.defineClass.goog.defineClass.abort_.error.executeNext_ (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2758:21)
at goog.async.run.processWorkQueue (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:124:15)
at process._tickCallback (node.js:368:9)
Error
at [object Object].ElementArrayFinder.applyAction_ (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\lib\element.js:380:21)
at [object Object].ElementArrayFinder.(anonymous function) [as takeScreenshot] (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\lib\element.js:78:17)
at [object Object].ElementFinder.(anonymous function) [as takeScreenshot] (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\lib\element.js:709:7)
at Object.<anonymous> (F:\tepez\tepez-front\src\app\form\e2e\validation.e2e.js:27:53)
at [object Object].promise.Promise.goog.defineClass.constructor (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:1056:7)
at new wrappedCtr (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\base.js:2468:26)
at [object Object].promise.ControlFlow.goog.defineClass.goog.defineClass.abort_.error.executeNext_.execute_ (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2776:14)
at [object Object].promise.ControlFlow.goog.defineClass.goog.defineClass.abort_.error.executeNext_ (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2758:21)
at goog.async.run.processWorkQueue (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:124:15)
at process._tickCallback (node.js:368:9)
From: Task: Run it in control flow
at [object Object].promise.Promise.goog.defineClass.invokeCallback_ (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:1337:14)
at [object Object].promise.ControlFlow.goog.defineClass.goog.defineClass.abort_.error.executeNext_.execute_ (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2776:14)
at [object Object].promise.ControlFlow.goog.defineClass.goog.defineClass.abort_.error.executeNext_ (C:/Users/tomyam/AppData/Roaming/npm/node_modules/protractor/node_modules/selenium-webdriver/lib/goog/../webdriver/promise.js:2758:21)
at goog.async.run.processWorkQueue (C:\Users\tomyam\AppData\Roaming\npm\node_modules\protractor\node_modules\selenium-webdriver\lib\goog\async\run.js:124:15)
at process._tickCallback (node.js:368:9)
```

I got the same error when I used `getWebElement().takeScreenshot()`.

Any ideas how to solve this?

**Comment by cnishina** [comment]
It appears taking a screenshot of a web element is a recent commit to selenium and uses `"version": "2.49.0-dev"` of selenium webdriver. When we upgrade from 2.48.0, we will add `takeScreenshot` to the [WEB_ELEMENT_FUNCTIONS](https://github.com/angular/protractor/blob/master/lib/element.js#L5) array.

**Comment by tepez** [comment]
Then maybe it's best to remove it from protractor's API documentation in the meantime.

http://www.protractortest.org/#/api?view=webdriver.WebElement.prototype.takeScreenshot

**Comment by Xotabu4** [comment]
+1 to upgrade to 2.49
Currently using additional library to crop images - that not cool at all.

**Comment by Xotabu4** [comment]
@cnishina, @tepez   Hi!
Fresh version of protractor was released, and now selenium version is 2.52 , so takeScreenshot can be added to WEB_ELEMENT_FUNCTIONS array?

**Comment by juliemr** [comment]
This is fixed for next release from the Protractor side, but note that many browser drivers still will not support element/screenshot

**Comment by vovan47** [comment]
@juliemr is it fixed now? and what browsers certainly support the feature? what about phantomjs? ;)

**Comment by juliemr** [comment]
Fixed now with 4.0 out.
